source/agora/common/ManagedDatabase.d

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+
+    Contains a managed wrapper around SQLite databases.
+
+    This class ensures that all D2SQLite databases handled by the thread
+    are cleanly destroyed after the thread shuts down.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.ManagedDatabase;
+
+import d2sqlite3.database;
+import d2sqlite3.sqlite3;
+
+/// Ditto
+public class ManagedDatabase
+{
+    /// ManagedDatabase managed by the current thread
+    private static Database*[] thread_dbs;
+
+    /// Close all database handles
+    static ~this ()
+    {
+        foreach (db; thread_dbs)
+            db.close();
+    }
+
+    /// Pointer to the database handle
+    private Database* database;
+
+    /// Subtype
+    public alias getDatabase this;
```

Comment:
I intend to replace this with just an `execute` wrapper which handles all the various SQLite error codes, and maybe calls an error delegate or something like that to make the calling code `nothrow`. But I'll add this to another PR.